apollo-cache-interceptor/src/commonMain/kotlin/com/apollographql/apollo/interceptor/cache/ApolloCacheInterceptor.kt

```diff
@@ -62,10 +106,14 @@ class ApolloCacheInterceptor<S>(private val store: S) : ApolloRequestInterceptor
         CacheHeaders.NONE,
         RealCacheKeyBuilder()
     )
-    val data = operation.parseData(rootRecord, request.customScalarAdapters, fieldValueResolver)
+    val data = operation.parseData(rootRecord, customScalarAdapters, fieldValueResolver)
 
-    return builder<D>(operation)
-        .data(data)
-        .build()
+    return ApolloResponse(
+        response = builder<D>(operation)
```

Comment:
not related to this PR, but does it make sense to have a builder for data classes in Kotlin? Data classes already kinda a builder with named arguments and default values, we probably should avoid using builders in new runtime?

Reply:
Good question. One advantage of `Builders` is that they "force" to use the return value. Take this exemple where we want to set the cache policy:

```
// Without builders, someone used to writing imperative/mutable code might be tempted to do that
val request = ApolloQueryRequest(HeroNameQuery())
request.fetchPolicy(NETWORK_ONLY)

apolloClient.query(request) // this doesn't use the NETWORK_ONLY policy because the return value isn't used above
```

Maybe it's just education but I feel like `Builders` are more boilerplate for us but ultimately better APIs for users

Reply:
hm, good point